src/track/track.cpp

```diff
@@ -952,8 +961,9 @@ void Track::importPendingCueInfosMarkDirtyAndUnlock(
             m_record.getMetadata().getSampleRate());
     const auto trackId = m_record.getId();
     QList<CuePointer> cuePoints;
-    cuePoints.reserve(m_importCueInfosPending.size());
-    for (const auto& cueInfo : m_importCueInfosPending) {
+    cuePoints.reserve(m_pCueInfoImporterPending->size());
+    for (const auto& cueInfo : m_pCueInfoImporterPending->importCueInfosWithCorrectTiming(
+                 getLocation(), m_streamInfo->getSignalInfo())) {
```

Comment:
I just noticed that we should better pass `const QString& fileType = m_record.getFileType()` here instead of `const QString& filePath = getLocation()`. The Importer does not need the full `filePath`, only the `fileType`.

Unfortunately the file type is by convention just a lowercase string and not type-safe. We could introduce a constant `kFileTypeMp3` or function `bool isFileTypeMp3(...)` to avoid literal strings in the Serato code.

Reply:
> I just noticed that we should better pass `const QString& fileType = m_record.getFileType()` here instead of `const QString& filePath = getLocation()`. The Importer does not need the full `filePath`, only the `fileType`.

We still need the full file path for mp3guessenc. Currently, the CoreAudio soundsource uses that for guessing the offsets.